### PR TITLE
Run datadog-agent pipeline on branch push

### PR DESCRIPTION
**Pull request by chouquette**
This will simplify testing our omnibus changes. https://datadoghq.atlassian.net/browse/APL-1767

This requires https://github.com/DataDog/datadog-agent/pull/20605 to be merged in order to provide the omnibus branch through the environment

### CODE REVIEW

**Review by Pythyu — APPROVED** [review]
put a few comment but LGTM

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+trigger-agent-build:
+  <<: *skip_stable_branches
+  stage: test
+  trigger:
+    project: datadog/datadog-agent
+    strategy: depend
+    branch: chouquette/allow_override_omnibus_config
+  variables:
+    OMNIBUS_SOFTWARE_VERSION: $CI_COMMIT_BRANCH
+    RUN_ALL_BUILDS: "true"
```
[the remark this answers is not in the document]
Comment (chouquette) [reply .gitlab-ci.yml:24]: I'd argue that we want to ensure that a change in the build recipes doesn't break anything, including on less frequently run tests.
This was a frequent request on some of the PRs I opened here or on omnibus-ruby so I enabled it here, but I can remove it if it doesn't change much in the coverage

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+      when: never
+    - when: always
+
+trigger-agent-build:
+  <<: *skip_stable_branches
+  stage: test
+  trigger:
+    project: datadog/datadog-agent
+    strategy: depend
+    branch: chouquette/allow_override_omnibus_config
```
[the remark this answers is not in the document]
Comment (chouquette) [reply .gitlab-ci.yml]: Removed :)

File: .gitlab-ci.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (chouquette) [reply .gitlab-ci.yml:23]: As discussed during syncup, this now enables all builds & all kitchen tests

**Review by Pythyu — APPROVED** [review]

**Pull request merged**